llpc/context/llpcCompiler.cpp

```diff
@@ -1259,9 +1277,49 @@ Result Compiler::buildPipelineInternal(Context *context, ArrayRef<const Pipeline
 
   context->setDiagnosticHandlerCallBack(nullptr);
 
+  if (result == Result::Success && cl::AddHashToELF) {
+    // Add two note entries:
+    //  1. Note name with "llpc_cache_hash" and note description with the cache hash used for the cache lookup.
+    //  2. Note name with "llpc_version" and note description with the LLPC version - both major and minor.
+    //     The LLPC version information will help us to understand the hash generation algorithm. We have to
+    //     use a correct hash algorithm for the cache lookup.
+    //
+    // For example, if the hash is "4EDBED25 ADF15238 B8C92579 423DA423" and the LLPC version is 45.4
+    // (the major version is 45=0x2D and the minor version is 4=0x04), two new note entries will be
+    //
+    //  Unknown(0)                (name = llpc_cache_hash  size = 16)
+    //        0:4EDBED25 ADF15238 B8C92579 423DA423
+    //  Unknown(0)                (name = llpc_version  size = 8)
+    //        0:0000002D 00000004
+    //
+    const uint32_t llpcVersion[2] = {LLPC_INTERFACE_MAJOR_VERSION, LLPC_INTERFACE_MINOR_VERSION};
+    auto hash = context->getPipelineContext()->getPipelineHashCodeWithoutCompact();
+    NoteEntry notes[] = {
+        {"llpc_cache_hash", ArrayRef<uint8_t>(hash.bytes)},
```

Comment:
To fit in with the conventional use of .note, the name should be the vendor "AMD", and the type should distinguish the note type within AMD's note type space.

As you, er, noted elsewhere, you'll then need to define the note types in llvm/include/llvm/BinaryFormat/ELF.h. If you give us a patch, we could put it in our local llvm branch for now (with an internal review from the other users of the AMDGPU back-end), and then you could upstream it into llvm.

Reply:
I prepared a PR for it: https://github.com/GPUOpen-Drivers/llvm-project/pull/2

I am not sure sending it as a PR to the repo is a right thing or not. Please let me know if I misunderstood what you meant.

Reply:
can I merge this  PR ?